src/components/player-info.js

```diff
@@ -7,29 +7,58 @@ import { AVATAR_TYPES } from "../assets/avatars/avatars";
  */
 AFRAME.registerComponent("player-info", {
   schema: {
-    displayName: { type: "string" },
     avatarSrc: { type: "string" },
     avatarType: { type: "string", default: AVATAR_TYPES.LEGACY }
   },
   init() {
+    this.displayName = null;
     this.applyProperties = this.applyProperties.bind(this);
+    this.updateDisplayName = this.updateDisplayName.bind(this);
+    this.applyDisplayName = this.applyDisplayName.bind(this);
+
+    this.isLocalPlayerInfo = this.el.id === "player-rig";
+    this.playerSessionId = null;
+
+    if (!this.isLocalPlayerInfo) {
+      NAF.utils.getNetworkedEntity(this.el).then(networkedEntity => {
+        this.playerSessionId = NAF.utils.getCreator(networkedEntity);
+        const playerPresence = window.APP.hubChannel.presence.state[this.playerSessionId];
+        if (playerPresence) {
+          this.displayName = playerPresence.metas[0].profile.displayName;
```

Comment:
maybe this and the two-liner on 51 should be turned into a function that receives the data structure at `playerPresence.metas[0]`